test/SILOptimizer/side_effects.sil

```diff
@@ -1281,3 +1281,64 @@ bb0(%0: $any Error):
   %r = tuple ()
   return %r : $()
 }
+
+sil @test_compute_deinit_barrier_callee_with_deinit_barrier : $@convention(thin) () -> @out Any {
+  [global: deinit_barrier]
+}
+
+
+sil @test_compute_deinit_barrier_callee_without_deinit_barrier : $@convention(thin) () -> @out Any {
+  [global: ]
+}
+
+// CHECK-LABEL: sil [readnone] @test_compute_deinit_barrier_callee_with_effect_attribute
+// CHECK-NEXT: [global: ]
+// CHECK-NEXT:  {{^[^[]}}
+sil [readnone] @test_compute_deinit_barrier_callee_with_effect_attribute : $@convention(thin) () -> @out Any {
+bb0(%0 : $*Any):
+%11 = tuple ()
+return %11
+}
```

Comment:
Please also add a negative test: a `[readonly]` function which _has_ a deinit barrier, e.g. because it's calling another function for which the body is not available.